manifests/install_systemd_unit.pp

```diff
@@ -2,15 +2,36 @@
 	$unit_options = {},				# Arguments to pass to the systemd [Unit] section of template
 	$service_options = {},			# Arguments to pass to the systemd [Service] section of template
 	$install_options = {},			# Arguments to pass to the systemd [Install] section of template
+	$timer_timer_options = {},		# Arguments to pass to the systemd [Timer] section of the timer template
+	$timer_unit_options = {},		# Arguments to pass to the systemd [Unit] section of the timer template
+	$timer_install_options = {},	# Arguments to pass to the systemd [Install] section of the timer template
 	$service_ensure = "running",	# State of the service after installed
-	$service_enable = true			# Enabled state of the service after installed
+	$service_enable = true,			# Enabled state of the service after installed
+	$path = "/etc/systemd/system"	# Path to systemd files
 ) {
-
 	::systemd::unit_file { "${name}.service":
-		content => template("${module_name}/systemd_service.erb")
-	} -> service { $name:
+		content => template("${module_name}/systemd_service.erb"),
+		path => $path,
+	}->
+	service { $name:
 		ensure => $service_ensure,
 		enable => $service_enable,
-		require => Exec['systemctl-daemon-reload'],
+		notify => Exec['systemctl-daemon-reload'],
+	}
+
+	if !empty($timer_timer_options) {
+		file { "${path}/${name}.timer":
+			content => template("${module_name}/systemd_timer.erb"),
+			owner   => 'root',
+			group   => 'root',
+			mode    => '0444',
+			notify => Exec["restart_${name}.timer"],
```

Comment:
you might need to notify `Exec['systemctl-daemon-reload']` so systemd reloads all configuration after adding a timer?

Reply:
also the timer is missing a service like the unit file, it should probably have one?

Reply:
It's a pity we can't use a newer upstream module, this works with timers by just naming the system resource with ".timer": https://github.com/camptocamp/puppet-systemd/blob/2.1.0/manifests/unit_file.pp#L7 :cry:

Reply:
There's a whole lot of other systemd types, maybe it should auto-detect the kind you're trying to make from the name (if it has ".timer" in it) like the newer upstream module version does, and alter the systemd unit template accordingly? otherwise if someone else came along and wanted another one of these types we'd have to introduce even more resources: https://github.com/camptocamp/puppet-systemd/blob/master/manifests/unit_file.pp#L7